analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -295,9 +297,15 @@ private static ProgramState InitLocals(ControlFlowBranch branch, ProgramState st
         branch.IsConditionalSuccessor ? condition == constraint : condition != constraint;
 
     private static bool IsReachable(ControlFlowRegion region, ExceptionState thrown) =>
-        region.Kind == ControlFlowRegionKind.Finally
+        CatchesAll(region)
         || thrown == ExceptionState.UnknownException
-        || region.ExceptionType.Is(KnownType.System_Object)       // catch when (condition)
```

Comment:
I find this comment about `catch when (condition)` important, as exception type `object` sounds weird by itself.